operator/docs/enhancements/retention_support.md

```diff
@@ -0,0 +1,218 @@
+
+## Motivation
+
+The Loki Operator manages `LokiStack` resources that consists of a set of Loki components for ingestion/quering and optionally a gateway microservice that ensures authenticated and authorized access to logs stored by Loki. 
+Retention in Loki has always been global for a cluster and deferred to the underlying object store. Since v2.3.0 Loki can handle retention through the Compactor component. Retention can be configured per tenant and per stream. These different retention configurations allow storage cost control and meet security and compliance requirements in a more granular way.
+A common use case for custom policies is to delete high-frequency logs earlier than other (low-frequency) logs.
+
+### Goals
+
+* The user can enable the retention via the `LokiStack` custom resource.
+* The user can declare per-tenant and global policies through the LokiStack custom resource. These are ordered by priority. 
+* The policies support time-based deletion of older logs.
+
+
+### Non-Goals
+
+* Stress-testing the compactor on each T-shirt-size with an overwhelming amount of retention rules
+
+## Proposal
+
+The following enhancement proposal describes the required API additions and changes in the Loki Operator to add support for configuring custom log retention per tenant
+https://grafana.com/docs/loki/latest/operations/storage/retention/
+https://grafana.com/docs/loki/latest/configuration/#compactor_config
+
+### API Extensions
+
+#### LokiStack Changes: Support for configuring log retention
+
+
+```go
+// LokiDuration defines the type for Prometheus durations.
+//
+// +kubebuilder:validation:Pattern:="((([0-9]+)y)?(([0-9]+)w)?(([0-9]+)d)?(([0-9]+)h)?(([0-9]+)m)?(([0-9]+)s)?(([0-9]+)ms)?|0)"
+type LokiDuration string
+
+// LokiStackSpec defines the desired state of LokiStack
+type LokiStackSpec struct {
+...
+    // Retention defines the spec for log retention
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    Retention *RetentionSpec `json:"retention,omitempty"`
+...
+}
+
+// RetentionSpec defines the spec for the enabling retention in the Compactor.
+type RetentionSpec struct {
+
+    // DeleteDelay defines Delay after which chunks will be fully deleted during retention
+    //
+    // +optional
+    // +kubebuilder:validation:optional
+    DeleteDelay LokiDuration `json:"deletedelay,omitempty"`
+}
+
+
+// LimitsTemplateSpec defines the limits  applied at ingestion or query path.
+type LimitsTemplateSpec struct {
+...
+    // RetentionLimits defines the configuration of the retention period.
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    RetentionLimits *RetentionLimitSpec `json:"retention,omitempty"`
+}
+
+// RetentionLimitSpec configures the reention period and retention stream
+type RetentionLimitSpec struct {
+    // Period defines the log retention period.
+    //
+    // +optional
+    // +kubebuilder:validation:Optional
+    Period LokiDurationntSpec `json:"period,omitempty"`
```

Comment:
The minimum retention period is `24h`. Maybe it is worth only configuring hours, days, months, years and then validate (via webhook) that the value is above 24h.